user-service/controller/user-controller.js

```diff
@@ -16,7 +16,7 @@ export const getUser = async (req, res) => {
   const { userId } = req
 
   const user = await _findUserById(userId, { password: 0 })
-  if (user.err) {
+  if (!user || user.err) {

```

Comment:
Error case I missed out previously. If no user with the given userId exists, the returned result is `null`, so we need to check before doing `user.err`